### PR TITLE
Build linux and darwin arm64 binary

### PR DESCRIPTION
**Pull request by wiget**
This allow using ls-lint on Linux run on arm64 CPU (Raspberry Pi4, Rock Pi4, AWS EC2 with AWS Graviton CPU (A1, T4g, C6g, M6g, R6g)).

### CODE REVIEW

**Comment by loeffel-io** [comment]
Thank you very much for this pull request @wiget - looks great

2 updates i would love to see: 

- We should also add the different darwin versions 
- We should also add a docker arm version

**Comment by loeffel-io** [comment]
closes #43

**Comment by wiget** [comment]
@loeffel-io If by different darwin version you mean version for Apple Silicon then this is planned for Go 1.16.

I will look at docker image for linux/arm64.

**Comment by loeffel-io** [comment]
Hey @wiget , crazy that darwin arm64 is not released yet

Alright 👍

**Comment by wiget** [comment]
@loeffel-io I finished Docker part too. Please review.

**Comment by wiget** [comment]
@loeffel-io anything missing or need to be adjusted before merge?

**Comment by wiget** [comment]
@loeffel-io I'm trying to bring [Renovate](https://github.com/renovatebot/renovate/) to arm64. As ls-lint is dependency of Renovate this PR is pre-requisite of my future work on the topic.

**Comment by loeffel-io** [comment]
Thanks @wiget, i currently try to dig into that - first of all: you must update the package.json version to release the new ls-lint version

File: .drone.yml [hunk 1]
```diff
@@ -50,18 +56,94 @@ steps:
...
     when:
       event: tag
 
+---
+kind: pipeline
+type: docker
+name: docker-linux-amd64
+depends_on:
+  - main
+trigger:
```
Comment (loeffel-io) [inline .drone.yml:69]: can we not just use 

``` 
trigger:
  branch:
  - master
  event:
  - tag
```

instead of the triggers?

Comment (wiget) [reply .drone.yml:69]: I'm not sure. Will need to test this.

Comment (wiget) [reply .drone.yml:69]: Looks like 
```yaml
trigger:
  event:
    - tag
 ```
works as expected.

**Comment by loeffel-io** [comment]
Thanks for the update but the branch is also very important ✌️

**Comment by wiget** [comment]
@loeffel-io I'm not sure if I understand correctly. Can you describe when Docker image should be build and published to Docker Hub? Than I will try to translate this into `.drone.yml`.

**Comment by wiget** [comment]
@loeffel-io any changes needed or this PR can be merged as-is?

**Comment by wiget** [comment]
@loeffel-io Any chance for this PR? 🙃

**Comment by wiget** [comment]
@loeffel-io Good news! golang 1.16 released with support for Apple Silicon. I added relevant changes to build darwin/arm64 version.

**Comment by loeffel-io** [comment]
Thank you very much! I think the biggest problem is that my ci works only for amd images - i need some more time to dig into this! sorry for the delay

**Comment by kxxoling** [comment]
Any update?

**Comment by caiofbpa** [comment]
Any news on merging this PR?

File: .drone.yml [hunk 2]
```diff
@@ -50,18 +57,94 @@ steps:
...
+  os: linux
+  arch: amd64
+
+steps:
+  - name: manifest
+    pull: always
+    image: plugins/manifest
+    settings:
+      auto_tag: true
+      ignore_missing: true
```
Comment (loeffel-io) [inline .drone.yml:145]: why do we want to ignore missing source images?

**Review by loeffel-io — CHANGES_REQUESTED** [review]

File: .drone.yml (hunk 2 above)
Comment (wiget) [reply .drone.yml:145]: I don't think is needed. I just copied this from some other project.

Comment (loeffel-io) [reply .drone.yml]: i removed it for now

**Comment by loeffel-io** [comment]
v1.10.0-beta.1 is online; please test the arm versions

**Comment by wiget** [comment]
Looks like it works on RockPi 4
```console
root@288e2632b2e9:/tmp/test# npx @ls-lint/ls-lint@1.10.0-beta.1
npx: installed 1 in 2.723s
2021/06/08 14:21:31 ./node_modules failed for rules: kebabcase
2021/06/08 14:21:31 ./node_modules/.bin failed for rules: kebabcase
2021/06/08 14:21:31 ./node_modules/@ls-lint failed for rules: kebabcase
root@288e2632b2e9:/tmp/test# uname -a
Linux 288e2632b2e9 5.10.21-rockchip64 #21.02.3 SMP PREEMPT Mon Mar 8 01:05:08 UTC 2021 aarch64 GNU/Linux
```

**Comment by loeffel-io** [comment]
Would love to hear more feedback about the v1.10.1-beta.1 before release ✌️

**Comment by padevone** [comment]
@loeffel-io, thanks for changes, but I think I have found a typo:

https://github.com/loeffel-io/ls-lint/blob/v1.10.0-beta.1/npm/bin/cli.js#L29
In this case is value `ls-lint-darwin-arm64`?

**Comment by loeffel-io** [comment]
@wiget

**Comment by wiget** [comment]
@padevone that's true. Are you able to submit PR with the fix? If not I will create PR later this week.

**Comment by loeffel-io** [comment]
would love to see a PR for this, currently at holidays @wiget

**Comment by caiofbpa** [comment]
Here: https://github.com/loeffel-io/ls-lint/pull/57

**Pull request merged**